### PR TITLE
Feat: First round of fuzzing tests

### PR DESCRIPTION
**Pull request by maciejka**
This is based on initial work from @LucasLvy. Adds addition/subtraction, multiplication, division, and formatting fuzzing test.

### CODE REVIEW

File: fuzz/felt/fuzz_targets/add.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    // Check a + b = a - (-b)
+    assert_eq!(a + b, a - (-b), "addition failed");
+
+    // Check a + a = a - (-a)
+    assert_eq!(a + a, a - (-a), "addition failed");
+    assert_eq!(b + b, b - (-b), "addition failed");
+
+    // Check a + a = 2 * a
+    assert_eq!(a + a, Felt::TWO * a, "doubling failed");
```
Comment (LucasLvy) [inline fuzz/felt/fuzz_targets/add.rs:35]: do it for b also

File: fuzz/felt/fuzz_targets/add.rs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    // Check a + b = b + a
+    assert_eq!(a + b, b + a, "commutativity failed");
+
+    // Check (a + b) + b = a + (b + b)
+    assert_eq!((a + b) + b, a + (b + b), "associativity failed");
+
+    // Check a + max = a - 1
+    assert_eq!(a + max, a - one, "overflow failed");
```
Comment (LucasLvy) [inline fuzz/felt/fuzz_targets/add.rs:44]: do it for b also

File: fuzz/felt/fuzz_targets/add.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    
+    // Check 0 - a = max - a + 1
+    assert_eq!(zero - a, max - a + one, "overflow failed");
```
Comment (LucasLvy) [inline fuzz/felt/fuzz_targets/add.rs:47]: do it for b

File: fuzz/felt/fuzz_targets/conversions.rs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+use starknet_types_core::felt::{Felt};
+
+fuzz_target!(|felt: Felt| {
+    assert_eq!(felt, Felt::from_bytes_be(&felt.to_bytes_be()), "to_bytes_be failed");
```
Comment (LucasLvy) [inline fuzz/felt/fuzz_targets/conversions.rs:6]: to bits also ?

File: fuzz/felt/fuzz_targets/conversions.rs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    assert_eq!(felt, Felt::from_bytes_le(&felt.to_bytes_le()), "to_bytes_le failed");
```
Comment (LucasLvy) [inline fuzz/felt/fuzz_targets/conversions.rs:7]: from dec string

File: fuzz/felt/fuzz_targets/conversions.rs [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    assert_eq!(felt, Felt::from_bytes_be_slice(&felt.to_bytes_be()), "to_bytes_be failed");
```
Comment (LucasLvy) [inline fuzz/felt/fuzz_targets/conversions.rs:8]: to digits

File: fuzz/felt/fuzz_targets/conversions.rs [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    assert_eq!(felt, Felt::from_bytes_le_slice(&felt.to_bytes_le()), "to_bytes_be failed");
```
Comment (LucasLvy) [inline fuzz/felt/fuzz_targets/conversions.rs:9]: to bigint (i probably missed some)

**Review by LucasLvy — CHANGES_REQUESTED** [review]
need to test double also

**Review by LucasLvy — CHANGES_REQUESTED** [review]
either separate add and sub or rename add to be add + sub (file names + fuzzing name)

**Review by LucasLvy — APPROVED** [review]

**Review by pefontana — APPROVED** [review]

**Comment by pefontana** [comment]
Nice one!

**Pull request merged**